[Improvement][Service] Modify ordinary users tenants

zombietramp opened the pull request:
1: Modify ordinary users and disable the function of selecting other tenants       Scenario: After the bank's internal administrator creates a new ordinary user and assigns a tenant. No other tenants can be selected after the new process

<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request

<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log

<!--*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--  -->

Commit:
1:修改普通用户，禁用选择其它租户的功能
2:增加流程定义sql语句不能有多条语句的提示，运行时候自动截取以;结尾的第一条语句
3:修改sql语句动态where 代替 where 1=1
4:修改template 不支持v-show命令

chengshiwen commented:
@zombietramp Please use English, and resolve the conflicts.
By the way, the v-show problem on template has been solved in pr #5145

Commit:
1:修改普通用户，禁用选择其它租户的功能 2:增加流程定义sql语句不能有多条语句的提示，运行时候自动截取以;结尾的第一条语句 3:修改sql语句动态where 代替 where 1=1 4:修改template 不支持v-show命令

Commit:
Merge remote-tracking branch 'origin/dev-local' into dev-local

# Conflicts:
#	dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml

CalvinKirs commented:
Thanks for your contribution. Your PR title does not seem to match the description.
Regarding multi-line SQL execution, can you describe your specific implementation?

ruanwenjun commented on dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml:
```diff
@@ -149,8 +150,7 @@
         join t_ds_project p on p.id=d.project_id
-        where 1 = 1
-        and t.is_sub_process = 0
+        where  and t.is_sub_process = 0
```
Should remove `and`

ruanwenjun commented on dolphinscheduler-common/sql/soft_version:
```diff
@@ -0,0 +1 @@
+32432423
```
ignore this file

ruanwenjun commented on dolphinscheduler-common/test/testFile.txt:
```diff
@@ -0,0 +1 @@
+正正正faffdasfasdfas
```
ignore this file

ruanwenjun commented on dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/UserMapper.xml:
```diff
@@ -19,7 +19,7 @@
 <mapper namespace="org.apache.dolphinscheduler.dao.mapper.UserMapper">
     <sql id="baseSql">
-       id, user_name, user_password, user_type, email, phone, tenant_id, create_time, update_time, queue, state
+       id, user_name, user_password, user_type, email, phone, tenant_id, create_time, update_time, queue, `state`
```
For the Sql style consistency, maybe it better not to make this modification ~~

ruanwenjun commented on dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java:
```diff
@@ -42,17 +42,17 @@
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
The use of import * is considered a bad practice

Commit:
1: Modify ordinary users and disable the function of selecting other tenants
      Scenario: After the bank's internal administrator creates a new ordinary user and assigns a tenant. No other tenants can be selected after the new process

2: Increase the process definition sql statement can not have multiple statement prompts, the first statement ending with; is automatically intercepted when running
    If there are multiple sql after sql, run error

3: Modify sql statement dynamic where instead of where 1=1

Commit:
[Improvement][Service] Modify ordinary users tenants
1: Modify ordinary users and disable the function of selecting other tenants       Scenario: After the bank's internal administrator creates a new ordinary user and assigns a tenant. No other tenants can be selected after the new process

Commit:
Merge remote-tracking branch 'origin/dev-local' into dev-local

songjianet commented:
Thank you for your contribution, but since this issue has not been dealt with for a long time, I will close it. If you still have problems in the future, you can re-submit a new issue.
